### ISSUE TITLE
Crash after installing/enabling oxidized-web

### ISSUE THREAD

**Issue by fhibler** [ytti/oxidized-web#84]
2016-01-07 10:52:36 - SyntaxError - /home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:9: syntax error, unexpected ':', expecting =>
'data-toggle': 'collapse',
              ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:9: syntax error, unexpected ',', expecting tSTRING_DEND
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:10: syntax error, unexpected ':', expecting tSTRING_DEND
'data-target': '#ox-nav',
              ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:10: syntax error, unexpected ',', expecting tSTRING_DEND
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected ':', expecting tSTRING_DEND
'aria-expanded': 'false')}>\n            <spa...
                ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected ')', expecting tSTRING_DEND
'aria-expanded': 'false')}>\n            <span class='...
                         ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected keyword_class, expecting keyword_do or '{' or '('
...se')}>\n            <span class='sr-only'>Toggle Navigation<...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected tCONSTANT, expecting keyword_do or '{' or '('
...ass='sr-only'>Toggle Navigation</span>\n            <span cl...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: unknown regexp options - pa
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected $undefined
...<span class='icon-bar'></span>\n            <span class='ico...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected keyword_class, expecting keyword_do or '{' or '('
.../span>\n            <span class='icon-bar'></span>\n        ...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected '<'
...       <span class='icon-bar'></span>\n            <span cla...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: unknown regexp options - pa
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:11: syntax error, unexpected $undefined
...<span class='icon-bar'></span>\n          </button>\n       ...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: unknown regexp option - g
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected $undefined
...ng_40px.png'))}>Oxidized</img>\n          </a>\n        </di...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: unknown regexp options - dv
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected $undefined
...          </a>\n        </div>\n        <div class='collapse...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected keyword_class, expecting keyword_do or '{' or '('
...     </div>\n        <div class='collapse navbar-collapse' i...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected tIDENTIFIER, expecting tSTRING_DEND
...s='collapse navbar-collapse' id='ox-nav'>\n          <ul cla...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected $undefined
... navbar-collapse' id='ox-nav'>\n          <ul class='nav nav...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected keyword_class, expecting keyword_do or '{' or '('
...='ox-nav'>\n          <ul class='nav navbar-nav'>\n         ...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:17: syntax error, unexpected $undefined
...   <ul class='nav navbar-nav'>\n            <li#{
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:21: syntax error, unexpected tSTRING_DEND, expecting '}'
_hamlout.attributes({}, nil, class: request.path_in...
                      ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:21: Can't assign to nil
_hamlout.attributes({}, nil, class: request.path_info == ...
                            ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:21: syntax error, unexpected ')', expecting tSTRING_DEND
...'/nodes/stats' ? 'active' : '')}>\n              <a#{_hamlou...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:22: syntax error, unexpected '}', expecting tSTRING_DEND
...href: url_for('/nodes/stats'))}>Stats</a>\n            </li>...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:22: unknown regexp option - l
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:22: syntax error, unexpected $undefined
...}>Stats</a>\n            </li>\n            <li#{_hamlout.ad...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:24: syntax error, unexpected '}', expecting tSTRING_DEND
... '/migration' ? 'active' : '')}>\n              <a#{_hamlout...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:25: syntax error, unexpected '}', expecting tSTRING_DEND
..., href: url_for('/migration'))}>Migration</a>\n            <...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:25: unknown regexp option - l
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:25: syntax error, unexpected $undefined
...gration</a>\n            </li>\n          </ul>\n          <...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: unknown regexp options - pa
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: syntax error, unexpected $undefined
...icon glyphicon-search'></span>\n              </button>\n   ...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: unknown regexp options - dv
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: syntax error, unexpected $undefined
... </button>\n            </div>\n          </form>\n        <...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: unknown regexp options - dv
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: syntax error, unexpected $undefined
...       </form>\n        </div>\n      </div>\n    </nav>\n  ...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: unknown regexp options - av
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: syntax error, unexpected $undefined
...div>\n      </div>\n    </nav>\n    <div class='container-fl...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: syntax error, unexpected keyword_class, expecting keyword_do or '{' or '('
...iv>\n    </nav>\n    <div class='container-fluid'>\n      #{...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:34: syntax error, unexpected $undefined
... <div class='container-fluid'>\n      #{_hamlout.adjust_tabs...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:39: syntax error, unexpected '}', expecting tSTRING_DEND
));}\n    </div>\n    <script#{_h...
    ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:39: unknown regexp options - crpt
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:39: syntax error, unexpected $undefined
...bootstrap.min.js'))}></script>\n    <script#{
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:40: syntax error, unexpected '}', expecting tSTRING_DEND
...ts/jquery.dataTables.min.js'))}></script>\n    <script#{
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:41: unknown regexp options - crpt
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:41: syntax error, unexpected $undefined
...les.bootstrap.js'))}></script>\n    <script#{
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:42: syntax error, unexpected '}', expecting tSTRING_DEND
...cripts/dataTables.colVis.js'))}></script>\n    <script#{
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:43: unknown regexp options - crpt
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:43: syntax error, unexpected $undefined
...ipts/oxidized.js'))}></script>\n  </body>\n</html>\n", -2, f...
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:43: unknown regexp options - htl
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:43: syntax error, unexpected $undefined
...></script>\n  </body>\n</html>\n", -2, false);
...                               ^
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:43: unterminated string meets end of file
/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/views/layout.haml:43: syntax error, unexpected end-of-input, expecting tSTRING_DEND
end;end;end;end
               ^:
	/home/oxidized/.gem/ruby/gems/tilt-1.4.1/lib/tilt/template.rb:264:in `class_eval'
	/home/oxidized/.gem/ruby/gems/tilt-1.4.1/lib/tilt/template.rb:264:in `compile_template_method'
	/home/oxidized/.gem/ruby/gems/tilt-1.4.1/lib/tilt/template.rb:240:in `compiled_method'
	/home/oxidized/.gem/ruby/gems/tilt-1.4.1/lib/tilt/template.rb:169:in `evaluate'
	/home/oxidized/.gem/ruby/gems/tilt-1.4.1/lib/tilt/haml.rb:24:in `evaluate'
	/home/oxidized/.gem/ruby/gems/tilt-1.4.1/lib/tilt/template.rb:103:in `render'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:822:in `render'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:831:in `block in render'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:831:in `catch'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:831:in `render'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:676:in `haml'
	/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/webapp.rb:235:in `out'
	/home/oxidized/.gem/ruby/gems/oxidized-web-0.6.0/lib/oxidized/web/webapp.rb:45:in `block in <class:WebApp>'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1610:in `call'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1610:in `block in compile!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:974:in `[]'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:974:in `block (3 levels) in route!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:993:in `route_eval'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:974:in `block (2 levels) in route!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1014:in `block in process_route'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1012:in `catch'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1012:in `process_route'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:972:in `block in route!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:971:in `each'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:971:in `route!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1084:in `block in dispatch!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `block in invoke'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `catch'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `invoke'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1081:in `dispatch!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:906:in `block in call!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `block in invoke'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `catch'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1066:in `invoke'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:906:in `call!'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:894:in `call'
	/home/oxidized/.gem/ruby/gems/rack-protection-1.5.3/lib/rack/protection/xss_header.rb:18:in `call'
	/home/oxidized/.gem/ruby/gems/rack-protection-1.5.3/lib/rack/protection/path_traversal.rb:16:in `call'
	/home/oxidized/.gem/ruby/gems/rack-protection-1.5.3/lib/rack/protection/json_csrf.rb:18:in `call'
	/home/oxidized/.gem/ruby/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'
	/home/oxidized/.gem/ruby/gems/rack-protection-1.5.3/lib/rack/protection/base.rb:49:in `call'
	/home/oxidized/.gem/ruby/gems/rack-protection-1.5.3/lib/rack/protection/frame_options.rb:31:in `call'
	/home/oxidized/.gem/ruby/gems/rack-1.6.0/lib/rack/nulllogger.rb:9:in `call'
	/home/oxidized/.gem/ruby/gems/rack-1.6.0/lib/rack/head.rb:13:in `call'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/show_exceptions.rb:21:in `call'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:181:in `call'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:2021:in `call'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1486:in `block in call'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1795:in `synchronize'
	/home/oxidized/.gem/ruby/gems/sinatra-1.4.6/lib/sinatra/base.rb:1486:in `call'
	/home/oxidized/.gem/ruby/gems/rack-1.6.0/lib/rack/urlmap.rb:66:in `block in call'
	/home/oxidized/.gem/ruby/gems/rack-1.6.0/lib/rack/urlmap.rb:50:in `each'
	/home/oxidized/.gem/ruby/gems/rack-1.6.0/lib/rack/urlmap.rb:50:in `call'
	/home/oxidized/.gem/ruby/gems/rack-1.6.0/lib/rack/builder.rb:153:in `call'
	/home/oxidized/.gem/ruby/gems/puma-2.11.1/lib/puma/server.rb:507:in `handle_request'
	/home/oxidized/.gem/ruby/gems/puma-2.11.1/lib/puma/server.rb:375:in `process_client'
	/home/oxidized/.gem/ruby/gems/puma-2.11.1/lib/puma/server.rb:262:in `block in run'
	/home/oxidized/.gem/ruby/gems/puma-2.11.1/lib/puma/thread_pool.rb:104:in `call'
	/home/oxidized/.gem/ruby/gems/puma-2.11.1/lib/puma/thread_pool.rb:104:in `block in spawn_thread'

**Comment by supertylerc** [comment]
Was this installed via RubyGems or via the repository?

**Comment by ytti** [comment]
Right now they should be same, with 0.6.0.

**Comment by supertylerc** [comment]
Thanks. Didn't realize there was a version bump. :D

I'll look at a fix today.

**Comment by ytti** [comment]
Is this same as ytti/oxidized#261

**Comment by supertylerc** [comment]
This looks like it's an issue specifically with Ruby 2.1 and earlier.  With Ruby 2.2.3 and 2.3.0, this is working as expected.  When I switch to 2.1.7, though, I'm seeing the same issue.

**Comment by ytti** [comment]
With commit provided, I'll release 0.6.1 immediately.

**Comment by supertylerc** [comment]
@ytti, see #86 for the fix.

@ytti, @fhibler, @alpern95: sorry for introducing the bug.  :(  I'll be sure to also test Ruby 2.0 and 2.1 in the future.

**Comment by supertylerc** [comment]
Just because:

```
tyler@hoth ~ $ ruby -v
ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-linux]
tyler@hoth ~ $ pry
[1] pry(main)> {'example-key': 'value'}
=> {:"example-key"=>"value"}

tyler@hoth ~ $ ruby -v
ruby 2.1.8p440 (2015-12-16 revision 53160) [x86_64-linux]
tyler@hoth ~ $ pry
[1] pry(main)> {'example-key': 'value'}
SyntaxError: unexpected '}', expecting end-of-input
```

**Comment by supertylerc** [comment]
See these Ruby feature requests (and related requests/bugs/commits) for when the changes that allow this to work in Ruby 2.2 and up:

https://bugs.ruby-lang.org/issues/4935
https://bugs.ruby-lang.org/issues/4276

**Comment by supertylerc** [comment]
#86 has been merged.  A new gem can be cut and pushed.

**Comment by ytti** [comment]
Realised I can't publish 0.6.1, after about 1200days of uptime neglect, my server is down, and I've moved few thousands kilometers from it, so might be bit challenging to get it up during the weekend.

**Comment by supertylerc** [comment]
@fhibler @alpern95:

Temporary workaround (until 0.6.1 is cut):

* Clone this repository (`git clone https://github.com/ytti/oxidized-web`)
* Build the gem (`rake build`)
* Install the gem (`gem install --no-ri --no-rdoc gems/oxidized-web-0.6.0.gem`)

You can leave off the `--no-ri` and `--no-rdoc` if you really want to.

**Comment by ytti** [comment]
fixed by ytti/oxidized-web@d63989e

oxidized-web has been released
